src/helics/shared_api_library/internal/api_objects.h

```diff
@@ -70,6 +75,15 @@ class CoreObject {
 /** get the CoreObject from a HelicsCore and verify it is valid*/
 CoreObject* getCoreObject(HelicsCore core, HelicsError* err) noexcept;
 
+/** object representing an app*/
+class AppObject {
+  public:
+    std::string type;  //!< the target of the query
+    std::shared_ptr<apps::App> app;
+    int index{-2};
+    int valid{0};
+};
+
```

Comment:
Why is this only necessary now?

Reply:
Quick answer is no one needed it before now.    There are some desired applications (by various programs) that require the connector to be able to execute in the same process as other applications, so it made the most sense to expose other apps in the same way as those will also likely be  needed in the future.